src/components/Icon/MucIcon.vue

```diff
@@ -0,0 +1,23 @@
+<template>
+  <svg
+    aria-hidden="true"
+    class="m-button__icon"
+    :style="{ color: color }"
+  >
+    <use :xlink:href="'#icon-' + icon" />
+  </svg>
+</template>
+
+<script setup lang="ts">
+defineProps<{
+  /**
+   * String of the icon to be displayed.
+   */
+  icon: string;
+
+  /**
+   * Optional color string for the icon.
```

Comment:
```suggestion
   * Optional css-color for the icon.
```

With your current Impl. I could also use `#121212` or `var(--my-color)`